graph/oval_graph.py

```diff
@@ -161,75 +161,52 @@ def change_tree_value(self, node_id, value):
     # Methods for interpreting oval tree with SigmaJS
 
     def _get_label(self):
-        return \
-            str(self.node_id).\
-            replace('xccdf_org.ssgproject.content_rule_', '').\
-            replace('oval:ssg-', '').\
-            replace(':def:1', '').\
-            replace(':tst:1', '').\
-            replace('test_', '')
-
-    def _create_node(self, x, y):
-        # print(self.evaluate_tree(),self.value)
+        if self.node_type == 'value':
+            return \
+                str(self.node_id).\
+                replace('xccdf_org.ssgproject.content_rule_', '').\
+                replace('oval:ssg-', '').\
+                replace(':def:1', '').\
+                replace(':tst:1', '').\
+                replace('test_', '')
+        else:
+            if str(self.node_id).startswith(
+                    'xccdf_org.ssgproject.content_rule_'):
+                return \
+                    str(self.node_id).\
+                    replace('xccdf_org.ssgproject.content_rule_', '')
+            return self.value
+
+    def _get_color_node(self):
         if self.value == 'true':
-            return {
-                'id': self.node_id,
-                'label': self._get_label(),
-                'url': 'null',
-                'text': 'null',
-                'title': self.node_id,
-                "x": x,
-                "y": y,
-                "size": 3,
-                "color": '#00ff00'}
+            return '#00ff00'
         elif self.value == 'false':
-            return {
-                'id': self.node_id,
-                'label': self._get_label(),
-                'url': 'null',
-                'text': 'null',
-                'title': self.node_id,
-                "x": x,
-                "y": y,
-                "size": 3,
-                "color": '#ff0000'}
+            return '#ff0000'
         else:
             if self.evaluate_tree() == 'true':
-                return {
-                    'id': self.node_id,
-                    'label': self.value,
-                    'url': 'null',
-                    'text': 'null',
-                    'title': self.node_id,
-                    "x": x,
-                    "y": y,
-                    "size": 3,
-                    "color": '#00ff00'
-                }
+                return '#00ff00'
```

Comment:
`self.evaluate_tree()` is called twice, which doesn't seem to be necessary, you can call it only once and save the returned value to a variable, and then use that variable in your conditions.